core/src/worker/workflow/history_update.rs

```diff
@@ -366,6 +400,46 @@ pub mod tests {
         assert_eq!(seq_2.last().unwrap().event_id, 8);
     }
 
+    #[tokio::test]
+    async fn history_ends_abruptly() {
```

Comment:
Do we have a test for WFT failed and timeout?
I was too lazy too look.

Reply:
Yes